lib/improver/utilities/spatial.py

```diff
@@ -230,6 +230,32 @@ def calculate_difference(self, cube, coord_axis):
             cube, coord_name, diff_along_axis)
         return diff_cube
 
+    @staticmethod
+    def gradient_from_diff(diff_cube, ref_cube, coord_axis):
+        """
+        Calculate the gradient along the x or y axis from differences between
+        adjacent grid squares.
+
+        Args:
+            diff_cube (Iris.cube.Cube):
+                Cube containing differences along the x or y axis
+            ref_cube (Iris.cube.Cube):
+                Cube with correct output dimensions
+            coord_axis (String):
+                Short-hand reference for the x or y coordinate, as allowed by
+                iris.util.guess_coord_axis.
+
+
+        Returns:
+            gradient (Iris.cube.Cube):
+                A cube of the gradients in the coordinate direction specified.
+        """
+        grid_spacing = np.diff(diff_cube.coord(axis=coord_axis).points)[0]
+        gradient = diff_cube.copy(data=abs(diff_cube.data / grid_spacing))
+        gradient = gradient.regrid(ref_cube, iris.analysis.Linear())
```

Comment:
This line is what is causing the problems when you compare the alpha values set as a float with those you're using as a grid. The linear regridding means that you are no longer guaranteed to be using values greater than 0 despite the abs(), the interpolation can lead to negative numbers. When this then gets scaled later on, all the sea points that have a gradient of zero are not at the bottom of the scaling (which is -0.005 for example). This means they are set to 0.498 or something similar, rather than 0.5. This explains the differences you see in the notebook. 

The linear regrid can remain here. Once you get rid of the abs() above, and introduce it in the process() part of OrographicAlphas, all your problems will go away.